elide-core/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -263,12 +306,319 @@ private Object fetchRelationship(Environment context) {
+        Set<PersistentResource> toReturn = new HashSet<>();
+
+        while(!toVisit.isEmpty()) {
+            Triplet triplet = toVisit.remove();
+            if(visited.contains(triplet)) {
+                continue;
+            } else {
+                visited.add(triplet);
+            }
+            Class<?> entityClass = getEntityClass(triplet.parentResource, dictionary, triplet.fieldName);
+            String idFieldName = getIdFieldName(dictionary, entityClass);
+            Optional<String> id = getId(triplet.input, idFieldName);
+            if(!id.isPresent()) {
+                setId(triplet.input, idFieldName);
+            }
+            /* first execute the given function with input data and then add the remaining relationships back to queue */
+            PersistentResource newParent = (PersistentResource) function.execute(triplet.input, getId(triplet.input,
+                    idFieldName).get(), requestScope, entityClass, triplet.fieldName, triplet.parentResource);
+            toReturn.add(newParent);
+            Map<String, Object> relationshipsOnly = stripAttributes(triplet.input, entityClass,
+                    dictionary);
+            /* loop over relationships */
+            for(Map.Entry<String, Object> relationship : relationshipsOnly.entrySet()) {
+                Boolean isToOne = newParent.getRelationshipType(relationship.getKey()).isToOne();
+                List<Map<String, Object>> entryToAdd;
+                if(isToOne) {
+                    entryToAdd = new ArrayList<>();
+                    entryToAdd.add((Map<String, Object>) relationship.getValue());
+                } else {
+                    entryToAdd = (List) relationship.getValue();
+                }
+                /* loop over each resource of the relationship */
+                for(Map<String, Object> entry : entryToAdd) {
+                    toVisit.add(new Triplet(entry, Optional.of(newParent), relationship.getKey()));
+                }
+            }
+        }
+        return toReturn;
+    }
+
+    /**
+     * update the relationship between {@param parent} and the resource loaded by given {@param id}
+     * @param dataField data input
+     * @param id id
+     * @param requestScope request scope
+     * @param entityClass entity class
+     * @param fieldName graphql field name
+     * @param parent parent resource
+     * @return {@link PersistentResource} object
+     */
+    private PersistentResource updateRelationship(Map<String, Object> dataField, String id,
+                                            RequestScope requestScope, Class<?> entityClass,
+                                            String fieldName, Optional<PersistentResource> parent) {
+        PersistentResource resource = PersistentResource.loadRecord(entityClass, id, requestScope);
+        parent.ifPresent(persistentResource ->
+                persistentResource.updateRelation(fieldName, Collections.singleton(resource)));
```

Comment:
This function cannot be called a single item at a time.  It is intended to be like our REPLACE operator.    It replaces what's there with what you are providing.  Each time you call this,  it will replace the previous element you put here with this new one.